src/test/java/com/google/sps/GameUtilTest.java

```diff
@@ -0,0 +1,245 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.ArrayList;
+import java.util.List;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import com.google.sps.utils.GameUtils;
+
+/** tests the game util functions */
+@RunWith(JUnit4.class)
+public final class GameUtilTest {
+
+  private static String userID1 = "1";
+  private static String userID2 = "2";
+
+  // helper variable allows the use of entities in testing 
+  private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  //  Given a user has no games, any game name would be valid
+  @Test
+  public void isValidGameUserHasNoGames() {
+
+    Entity userEntity2 = new Entity("user");
+    ArrayList<String> gameNames2 = new ArrayList<>();
+
+    userEntity2.setProperty("gameNames",gameNames2);
+    userEntity2.setProperty("userID",userID2);
+
+    boolean actual = GameUtils.IsValidGameName("game1", userEntity2);
+
+    Assert.assertEquals(true, actual);
+  }
+
+  // give IsValidGameName a game name the user already has 
+  @Test
+  public void isValidGameUserHasDuplicateGameName() {
+
+    Entity userEntity1 = new Entity("user");
+    ArrayList<String> gameNames1 = new ArrayList<>();
+
+    gameNames1.add("game1");
+    gameNames1.add("game2");
+    gameNames1.add("Game3");
+    userEntity1.setProperty("gameNames",gameNames1);
+    userEntity1.setProperty("userID",userID1);
+
+    boolean actual = GameUtils.IsValidGameName("game1", userEntity1);
+
+    Assert.assertEquals(false, actual);
+  }
+
+  // The user has games, but the given game name is new
+  @Test
+  public void isValidGameNewGameName() {
+
+    Entity userEntity1 = new Entity("user");
+    ArrayList<String> gameNames1 = new ArrayList<>();
+    
+    gameNames1.add("game1");
+    gameNames1.add("game2");
+    gameNames1.add("Game3");
```

Comment:
why is this one capital?

Reply:
oops missed that. It's changed.

Reply:
Did you push the change? It still shows up as upper case for me.

Reply:
I did 1/3 apparently, those should all be fixed now.